File: backend/app/schemas/welfare.py
from __future__ import annotations

from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class WelBenefitTypeItem(BaseModel):
    id: int
    code: str
    name: str
    module_path: str
    is_deduction: bool
    pay_item_code: Optional[str]
    is_active: bool
    sort_order: int
    created_at: datetime
    updated_at: datetime


class WelBenefitTypeListResponse(BaseModel):
    items: list[WelBenefitTypeItem]
    total_count: int
    page: int
    limit: int


class WelBenefitRequestItem(BaseModel):
    id: int
    request_no: str
    benefit_type_code: str
    benefit_type_name: str
    employee_no: str
    employee_name: str
    department_name: str
    status_code: str
    requested_amount: int
    approved_amount: Optional[int]
    payroll_run_label: Optional[str]
    description: Optional[str]
    requested_at: datetime
    approved_at: Optional[datetime]
    created_at: datetime
    updated_at: datetime


class WelBenefitRequestListResponse(BaseModel):
    items: list[WelBenefitRequestItem]
    total_count: int
    page: int
    limit: int


class WelBenefitTypeRowInput(BaseModel):
    id: Optional[int] = None
    code: str
    name: str
    module_path: str
    is_deduction: bool = False
    pay_item_code: Optional[str] = None
    is_active: bool = True
    sort_order: int = 0
    _status: str = "clean"


class WelBenefitTypeBatchRequest(BaseModel):
    items: list[WelBenefitTypeRowInput]


class WelBenefitTypeBatchResponse(BaseModel):
    created: int
    updated: int
    deleted: int


class WelBenefitRequestCreateRequest(BaseModel):
    benefit_type_code: str
    requested_amount: int
    description: Optional[str] = None


class WelBenefitRequestApproveRequest(BaseModel):
    approved_amount: int
    note: Optional[str] = None


class WelBenefitRequestRejectRequest(BaseModel):
    reason: Optional[str] = None


class WelBenefitRequestActionResponse(BaseModel):
    item: WelBenefitRequestItem

File: frontend/src/types/welfare.ts
export type WelBenefitTypeItem = {
  id: number;
  code: string;
  name: string;
  module_path: string;
  is_deduction: boolean;
  pay_item_code: string | null;
  is_active: boolean;
  sort_order: number;
  created_at: string;
  updated_at: string;
};

export type WelBenefitTypeListResponse = {
  items: WelBenefitTypeItem[];
  total_count: number;
  page: number;
  limit: number;
};

export type WelBenefitRequestItem = {
  id: number;
  request_no: string;
  benefit_type_code: string;
  benefit_type_name: string;
  employee_no: string;
  employee_name: string;
  department_name: string;
  status_code: string;
  requested_amount: number;
  approved_amount: number | null;
  payroll_run_label: string | null;
  description: string | null;
  requested_at: string;
  approved_at: string | null;
  created_at: string;
  updated_at: string;
};

export type WelBenefitRequestListResponse = {
  items: WelBenefitRequestItem[];
  total_count: number;
  page: number;
  limit: number;
};
